quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/TimefoldProcessor.java

```diff
@@ -321,36 +317,29 @@ private void assertSolverConfigSolutionClasses(IndexView indexView, Map<String,
-        if (annotationInstanceCollection.size() > 1 && solverConfigMap.size() == 1) {
+        var annotationInstanceList = getAllConcreteSolutionClasses(indexView);
+        var firstConfig = solverConfigMap.values().stream().findFirst().orElse(null);
+        if (annotationInstanceList.size() > 1 && solverConfigMap.size() == 1 && firstConfig != null
+                && firstConfig.getSolutionClass() == null) {
             throw new IllegalStateException("Multiple classes (%s) found in the classpath with a @%s annotation.".formatted(
-                    convertAnnotationInstancesToString(annotationInstanceCollection), PlanningSolution.class.getSimpleName()));
+                    convertAnnotationInstancesToString(annotationInstanceList), PlanningSolution.class.getSimpleName()));
         }
-        // Multiple classes and at least one solver config does not specify the solution class
-        List<String> solverConfigWithoutSolutionClassList = solverConfigMap.entrySet().stream()
+        // Multiple classes and at least one solver config do not specify the solution class
+        // We do not fail if all configurations define the solution,
+        // even though there are additional "unused" solution classes in the classpath.
+        var unconfiguredSolverConfigList = solverConfigMap.entrySet().stream()
                 .filter(e -> e.getValue().getSolutionClass() == null)
                 .map(Map.Entry::getKey)
                 .toList();
-        if (annotationInstanceCollection.size() > 1 && !solverConfigWithoutSolutionClassList.isEmpty()) {
+        if (annotationInstanceList.size() > 1 && !unconfiguredSolverConfigList.isEmpty()) {
             throw new IllegalStateException(
                     """
                             Some solver configs (%s) don't specify a %s class, yet there are multiple available (%s) on the classpath.
                             Maybe set the XML config file to the related solver configs, or add the missing solution classes to the XML files,
                             or remove the unnecessary solution classes from the classpath."""
-                            .formatted(String.join(", ", solverConfigWithoutSolutionClassList),
+                            .formatted(String.join(", ", unconfiguredSolverConfigList),
                                     PlanningSolution.class.getSimpleName(),
-                                    convertAnnotationInstancesToString(annotationInstanceCollection)));
-        }
-        // Unused solution classes
-        List<String> unusedSolutionClassList = annotationInstanceCollection.stream()
-                .map(planningClass -> planningClass.target().asClass().name().toString())
-                .filter(planningClassName -> solverConfigMap.values().stream().filter(c -> c.getSolutionClass() != null)
-                        .noneMatch(c -> c.getSolutionClass().getName().equals(planningClassName)))
-                .toList();
```

Comment:
I see that a test for this is being removed. Why?
Arguably, unused solution classes could point to a user error, and the user ought to know.